osm2gtfs/tests/core/tests_creator_factory.py

```diff
@@ -0,0 +1,116 @@
+                    ' " doesn\'t fit the naming convention.')
+
+                # Check for existing configuration file
+                config_file_path = path + selector + "/config.json"
+                self.assertTrue(
+                    os.path.isfile(config_file_path),
+                    'No correctly named configuration file found for: ' +
+                    selector)
+
+                # Check whether selector in config file is correctly
+                config = Configuration.load_config_file(open(config_file_path))
+                self.assertEqual(
+                    selector, config['selector'],
+                    "The selector (" + config['selector'] +
+                    ") in config file is not equal to the directory name: " +
+                    selector)
+
+                # Loop through available files in creator directories and check
+                # whether they are correctly named
+                for inner_subdir, inner_dirs, inner_files in os.walk(
+                        path + selector):
+
+                    for filename in inner_files:
+                        # Check the Python files (and ignore the others)
+                        if filename.endswith(
+                                '.py') and not filename == "__init__.py":
+
+                            # Check if selector has been properly applied to
+                            # file name
+                            self.assertTrue(
+                                filename.endswith(selector + '.py'),
+                                "This file is not well named: " + filename)
+
+                            # Check for valid creators
+                            split_filename = filename.split("_")
+                            if split_filename[0] == "feed":
+                                split_filename[0] = split_filename[0] + split_filename.pop(1)
+                            self.assertTrue(
+                                split_filename[0] in creators,
+                                "This file is not well named: " + filename)
+                            self.assertEqual(
+                                split_filename[1], "creator",
+                                "This file is not well named: " + filename)
+
+                            # Try to import the content of the creator module
+                            correct_class_name_state = True
+                            try:
+                                module = importlib.import_module(
+                                    ".creators." + selector + "." + filename[:-3],
+                                    package="osm2gtfs")
+                            except ImportError:
+                                correct_class_name_state = False
+
+                            # Snake to camel case to test for correct class names
+                            classname = ''.join(x.capitalize() or
+                                                '_' for x in filename[:-3].split('_'))
```

Comment:
Not sure if this is because of my lack of understanding, but why exactly did you use this `or '_'` part?

Reply:
I took it from [this source](https://www.w3resource.com/python-exercises/re/python-re-exercise-37.php), but it works also with out the `or '_'` part. So, I took it out.

Reply:
Please don't copy code whose copyright isn't yours into the project.

Reply:
Thanks for the reminder! You are very right!! I was wrongly assuming a public domain permission, which is not the case (only Creative Commons licensed). 

Anyway, with the latest changes it is not coming from any source, but rather a common knowledge based on a web research, the used code passage as it is - adapted to our code - can be found in numerous public places and doesn't belong to one creator.